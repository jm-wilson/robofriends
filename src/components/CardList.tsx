import React from 'react';
import Card from './Card';
import { IRobot } from '../containers/App';

interface ICardListProps {
  robots: IRobot[];
}

const CardList: React.FC<ICardListProps> = ({ robots }): JSX.Element => {
  return !robots.length ? (
    <div></div>
  ) : (
    <div>
      {robots.map((user, i) => {
        return <Card key={i} id={robots[i].id} name={robots[i].name} email={robots[i].email} />;
      })}
    </div>
  );
};

export default CardList;
