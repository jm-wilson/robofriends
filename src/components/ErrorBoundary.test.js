import { shallow } from 'enzyme';
import React from 'react';
import CardList from './CardList';
import ErrorBoundry from './ErrorBoundary';

it('should render ErrorBoundary component', () => {
  const mockRobots = [
    {
      id: 1,
      name: 'John Snow',
      email: 'john@gmail.com',
    },
  ];
  expect(
    shallow(
      <ErrorBoundry>
        <CardList robots={mockRobots} />
      </ErrorBoundry>,
    ),
  ).toMatchSnapshot();
});

it('should safely catch an error', () => {
  const wrapper = shallow(
    <ErrorBoundry>
      <CardList />
    </ErrorBoundry>,
  );

  const error = new Error('OH NO');
  wrapper.first().simulateError(error);

  expect(wrapper).toMatchSnapshot();
});
