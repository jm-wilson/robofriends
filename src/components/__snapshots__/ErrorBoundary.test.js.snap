// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ErrorBoundary component 1`] = `
<CardList
  robots={
    Array [
      Object {
        "email": "john@gmail.com",
        "id": 1,
        "name": "John Snow",
      },
    ]
  }
/>
`;

exports[`should safely catch an error 1`] = `
<h1>
  Oooops. That is not good
</h1>
`;
